bvm/ballerina-core/src/main/java/org/ballerinalang/bre/bvm/BVMExecutor.java

```diff
@@ -148,7 +149,11 @@ public static void executeResource(ProgramFile programFile, FunctionInfo resourc
         ObserveUtils.startResourceObservation(strand, observerContext);
 
         BVMScheduler.stateChange(strand, State.NEW, State.RUNNABLE);
-        BVMScheduler.schedule(strand);
+        if (ConfigRegistry.getInstance().getAsBoolean("b7a.http.worker.nonblocking")) {
```

Comment:
This is wrong IMO, you should do this in http level, in ballerina core level there should be two API methods exposed to run in same thread or to submit to a pool. Can we please correct this?